hedera-node/hedera-app/src/main/java/com/hedera/node/app/bbm/scheduledtransactions/ScheduledTransaction.java

```diff
@@ -0,0 +1,116 @@
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+
+@SuppressWarnings("java:S6218") // "Equals/hashcode methods should be overridden in records containing array fields"
+record ScheduledTransaction(
+        long number,
+        @NonNull Optional<JKey> adminKey,
+        @Nullable String memo,
+        boolean deleted,
+        boolean executed,
+        boolean calculatedWaitForExpiry,
+        boolean waitForExpiryProvided,
+        @Nullable EntityId payer,
+        EntityId schedulingAccount,
+        RichInstant schedulingTXValidStart,
+        @Nullable RichInstant expirationTimeProvided,
+        @Nullable RichInstant calculatedExpirationTime,
+        @Nullable RichInstant resolutionTime,
+        byte[] bodyBytes,
+        @Nullable TransactionBody ordinaryScheduledTxn,
+        @Nullable SchedulableTransactionBody scheduledTxn,
+        @Nullable List<byte[]> signatories) {
+
+    static ScheduledTransaction fromMod(@NonNull final Schedule value) throws InvalidKeyException {
+        return new ScheduledTransaction(
+                value.scheduleId().scheduleNum(),
+                value.adminKey() != null ? Optional.of(JKey.mapKey(value.adminKey())) : Optional.empty(),
+                value.memo(),
+                value.deleted(),
+                value.executed(),
+                // calculatedWaitForExpiry is the same as waitForExpiryProvided;
+                // see ScheduleVirtualValue::from` - to.calculatedWaitForExpiry = to.waitForExpiryProvided;
+                value.waitForExpiry(),
+                value.waitForExpiry(),
+                entityIdFrom(value.payerAccountId().accountNum()),
+                entityIdFrom(value.schedulerAccountId().accountNum()),
+                RichInstant.fromJava(Instant.ofEpochSecond(
+                        value.scheduleValidStart().seconds(),
+                        value.scheduleValidStart().nanos())),
+                RichInstant.fromJava(Instant.ofEpochSecond(value.providedExpirationSecond())),
+                RichInstant.fromJava(Instant.ofEpochSecond(value.calculatedExpirationSecond())),
+                RichInstant.fromJava(Instant.ofEpochSecond(
+                        value.resolutionTime().seconds(), value.resolutionTime().nanos())),
+                PbjConverter.fromPbj(value.originalCreateTransaction()).toByteArray(),
+                PbjConverter.fromPbj(value.originalCreateTransaction()),
+                PbjConverter.fromPbj(value.scheduledTransaction()),
+                value.signatories().stream()
+                        .map(ScheduledTransaction::toPrimitiveKey)
+                        .toList());
+    }
+
+    static ScheduledTransaction fromMono(@NonNull final ScheduleVirtualValue scheduleVirtualValue) {
+        return new ScheduledTransaction(
+                scheduleVirtualValue.getKey().getKeyAsLong(),
+                scheduleVirtualValue.adminKey(),
+                scheduleVirtualValue.memo().orElse(""),
```

Comment:
Nit: it might be helpful to have something like `.orElse("<EMPTY>");` here. Thoughts?

Reply:
Make sense to me! Updated 👍